### ISSUE TITLE
Diffs didn't match during apply: ignore_changes + referencing resources

### ISSUE THREAD

**Issue by Fodoj** [hashicorp/terraform#6746]
Given: two instances, second references first one inside metadata. 

_Rough example, but you will get the idea:_

```
resource "openstack_compute_instance_v2" "box-1" {
  name = "box-1"
  availability_zone = "${var.availability_zone}"
  image_id = "${var.image_id}"
  flavor_name = "m1.medium"
  key_pair = "${var.key_pair}"
  network {
    uuid = "${var.network_id}"
    fixed_ip_v4 = "192.168.0.21"
  }
  lifecycle {
    ignore_changes = ["user_data", "image_id"]
  }
}

resource "openstack_compute_instance_v2" "box-2" {
  name = "box-2-${count.index}"
  availability_zone = "${var.availability_zone}"
  image_id = "${var.image_id}"
  flavor_name = "m1.medium"
  key_pair = "${var.key_pair}"
  network {
    uuid = "${var.network_id}"
  }
  metadata {
    box_1_ip = "${openstack_compute_instance_v2.box-1.access_ip_v4}"
    random_var = "blah"
  }
  count = 2
  lifecycle {
    ignore_changes = ["user_data", "image_id"]
  }
}
```
1. Create with terraform apply
2. Change image_id
3. Terraform plan

```
~ openstack_compute_instance_v2.box-2.0
    metadata.box_1_ip :            "192.168.0.21" => ""
    metadata.random_var :            "blah" => ""
```

Wait, what, why?  Nothing was supposed to be changed, metadata is the same and metadata didnt change inside Openstack. 

Okay, anyways:
1. Terraform apply again

```
Error applying plan:

2 error(s) occurred:

* openstack_compute_instance_v2.box-2.0: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

    Terraform Version: 0.7.0
    Resource ID: openstack_compute_instance_v2.amadeus-ns-slave.0
    Mismatch reason: diff: Destroy; old: false, new: true

# Lots of lines, showing that "plan" promised to nullify metadata and then during apply it didnt happen and blah blah errors everywhere
```

Now we are doomed. because seems like ignore_changes doesnt work!

**Comment by jtopjian** [comment]
@Fodoj One reason that the metadata would change is because changing the `image_id` causes `ForceNew` to trigger. Though the _values_ of the metadata are the same, new resources are being created on both sides.

**Comment by Fodoj** [comment]
@jtopjian which should not happen at all (I mean, no new resources should be created), because ignore_changes ignores changes to image_id

**Comment by jtopjian** [comment]
Yup, that's a good point. Sounds like it's more of a core issue, I think?

**Comment by Fodoj** [comment]
It is definitely a core issue

**Comment by mitchellh** [comment]
This is fixed in the latest release (0.7.8).
